### PR TITLE
feat: improve `switch` output

### PR DESCRIPTION
**Pull request by evilebottnawi**
Based on https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L1631. Also all tests were adapted.

P.S. We don't break anything in switch statement :star:

### CODE REVIEW

File: tests/switch/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -107,15 +301,17 @@ switch ($expr) {
...
-) {
+switch ($longLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongVariable) {
 }
 
-switch (
-    $longLongLongLongLongLongLongVariable &&
-    $longLongLongLongLongLongLongVariable
-) {
+switch ($longLongLongLongLongLongLongVariable &&
+    $longLongLongLongLongLongLongVariable) {
```
Comment (evilebottnawi) [inline tests/switch/__snapshots__/jsfmt.spec.js.snap]: [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBnA7gSxmAFgAgAoASAG2gHMAZSmqa2x+uigNQEMAnTdgI1Lj4AZEIA6UfGSYNm0liw7c+A4WJj4Nk8rJ0y98yop784ASnzAAvvhAAaEBAAOMTNFTJQXThHQAFLgjuKOwAbhCYACZ2IOyoMMgAZuykqHD2vJzsYADWcDAAyo5ZmPTIMJwArmkgERBgicmp9iWpnDC+mRQAtuwNKdUAVqgAHgBCmTl5+exdcFQlcH1NDhUwjqsATEvVRZytyCC8fACe2tGO3LAA6pEwuMgAHAAM9hcQqVeZjgcXcK0hi3snDgAEcKphgR12N1ekgkv17KkupgypVqqgShQBABFCoQeDbewwPg3CJ3ZAbImZTCkTEAYQgXR6Byg0EBIAqqQAKnwgvDUpZLEA).

Should we do as `js`?

Comment (evilebottnawi) [reply tests/switch/__snapshots__/jsfmt.spec.js.snap]: @vjeux it is bug in `prettier` or expected behavior? Don't found issues around this.

Comment (vjeux) [reply tests/switch/__snapshots__/jsfmt.spec.js.snap]: Looks like we're not applying the same treatment for switch as for `if/while/...`. This is not intended. `switch` is very rarely used in js so it probably never come up in real code.

Comment (vjeux) [reply tests/switch/__snapshots__/jsfmt.spec.js.snap]: If you can send a fix to prettier that would be very welcome :)

Comment (evilebottnawi) [reply tests/switch/__snapshots__/jsfmt.spec.js.snap]: @vjeux yep, i will send PR :+1:

**Pull request merged**